### DISCUSSION TITLE
Pydantic schema conversion for nested jsons

### DISCUSSION THREAD

**Discussion by PranathiSG** [pydantic/pydantic#6127] [Question] [answered]
Hi, I'm using FastAPI with pydantic for schema validation. 

In my fastapi application, I'm recieveing data from external source in the below mentioned format.

example_list1 = [ ['11','customer1','tran1','item1','itemcd1','discount1','discountcoupon','cash1','cash2'], ['11','customer1','tran1','item1','itemcd1','discount2','discountcoupon','cash1','cash2'], ['11','customer1','tran1','item1','itemcd2','discount1','discountcoupon','cash1','cash2'], ['12','customer1','tran2','item1','itemcd1','discount1','discountcoupon','cash1','cash2'], ['11','customer1','tran1','item1','itemcd1','discount1','discountcoupon2','cash1','cash2'],
['12','customer1','tran2','item1','itemcd1','discount1','discountcoupon2','cash1','cash2'] ]

The required output schema should be in below format:

final_output = [
   {
      "id":"11",
      "custname":"customer1",
      "tranname":"tran1",
      "itemdetail":[
         {
            "seqno":"item1",
            "itemcd":"itemcd1"
         },
         {
            "seqno":"item1",
            "itemcd":"itemcd2"
         }
      ],
      "discountdetail":[
         {
            "seqno":"discount1",
            "dname":"discountcoupon"
         },
         {
            "seqno":"discount2",
            "dname":"discountcoupon"
         },
         {
            "seqno":"discount1",
            "dname":"discountcoupon2"
         }
      ],
      "tenderdetail":[
         {
            "seqno":"cash1",
            "type":"cash2"
         }
      ]
   },
   {
      "id":"12",
      "custname":"customer1",
      "tranname":"tran2",
      "itemdetail":[
         {
            "seqno":"item1",
            "itemcd":"itemcd1"
         }
      ],
      "discountdetail":[
         {
            "seqno":"discount1",
            "dname":"discountcoupon"
         },
         {
            "seqno":"discount1",
            "dname":"discountcoupon2"
         }
      ],
      "tenderdetail":[
         {
            "seqno":"cash1",
            "type":"cash2"
         }
      ]
   }
]

Is there any way to achieve final_output using pydantic ? Traditional python approach are causing performance issues since the number of data and data size in real time is much higher.

**Comment by chrisK824** [comment, marked as answer]
Hey @PranathiSG , I'm not sure what you are asking is possible, Pydantic is to validate and trim your objects based on declared schema and not to transform it somehow magically. I think you cannot avoid the format manipulation.
